src/snapping/index.js

```diff
@@ -148,23 +337,21 @@ class Snapping {
       return;
     }
 
-    const bufferLayerIds = this.bufferLayers.map(getBufferLayerId);
-    const selectedFeature = this.store.ctx.api.getSelected().features[0];
-    const { point: mousePosition } = e;
-    const { x, y } = mousePosition;
+    const {
+      point: { x, y },
+    } = e;
 
-    const featureFilter = getFeatureFilter(
-      selectedFeature,
-      this.snappedFeature,
-      mode
-    );
+    let snapToFeature;
+
+    // avoid snapping points to points
+    if (this._isLineDraw()) {
+      snapToFeature = this._getClosestPoint(x, y);
+    }
+
+    if (!snapToFeature) {
+      snapToFeature = await this._getClosestLineStringOrPolygon(x, y);
```

Comment:
So Im curious about this. This is getting called possibly ever 16 ms right? Isn't that too much even if its fast as hell?

also pattern-wise, when snapping to points we do the thing where we just use the mapbox geom until an event signifies the draw end, then deal with it. Is a similar pattern not available here?

 or is the darn thing so fast that that really would be unnecessary. 
 But if thats fast, why would doing it for points be any slower?
 :exploding_head:

Reply:
I just set it back to 100 ms for now. The reason the point pattern isn't available here is that the linestring coordinates get really weird when represented by mapbox, sometimes its even a multilinestring (whereas a point is always a point). So we can use the mapbox point to be a decent representation of the final point, but we can't do that with lines